src/views/CountryRiskWatch/MultiMonthSelectInput/index.tsx

```diff
@@ -0,0 +1,162 @@
+
+                    const truthyValueStartIndex = prevValueList.findIndex(Boolean);
+                    const newValueList = [...prevValueList];
+                    const lengthDiff = Math.abs(month - truthyValueStartIndex);
+                    newValueList.splice(
+                        Math.min(truthyValueStartIndex, month),
+                        lengthDiff,
+                        ...Array(lengthDiff + 1).fill(true),
+                    );
+                    const maxIndex = Math.max(truthyValueStartIndex, month) + 1;
+                    const remaining = newValueList.length - maxIndex;
+                    newValueList.splice(
+                        maxIndex,
+                        remaining,
+                        ...Array(remaining).fill(false),
+                    );
+                    newValueList.splice(12, 1, false);
+
+                    return listToMap(
+                        newValueList,
+                        (_, key) => key,
+                        (currentValue) => currentValue,
+                    );
+                },
+                name,
+            );
+        },
+        [onChange, name],
+    );
+
+    return (
+        <div className={_cs(styles.multiMonthSelectInput, className)}>
+            <div className={styles.monthList}>
+                {keyList.map(
+                    (key) => {
+                        date.setMonth(key);
+                        const monthName = date.toLocaleString(
+                            navigator.language,
+                            { month: 'short' },
+                        );
+
+                        return (
+                            <RawButton
+                                key={key}
+                                className={_cs(styles.month, value?.[key] && styles.active)}
+                                name={key}
+                                onClick={handleClick}
+                            >
+                                <span className={styles.name}>
+                                    {monthName}
+                                </span>
+                                <span className={styles.visualQue}>
+                                    <span className={styles.startBorder} />
+                                    <span className={styles.circle} />
+                                    <span className={styles.endBorder} />
+                                </span>
+                            </RawButton>
+                        );
+                    },
+                )}
+            </div>
+            <div className={styles.separator} />
+            <RawButton
+                name={12}
+                onClick={handleClick}
+                className={_cs(styles.month, value?.[12] && styles.active)}
+            >
+                {/* FIXME: use translation */}
+                <span className={styles.name}>
+                    Yearly Average
+                </span>
+            </RawButton>
```

Comment:
Why does this button do? Why is the value 12?

Reply:
As the name suggest, the button is supposed to incorporate the Yearly average. I'm not sure what do you mean by `Why does this button do?`. This button do because this button do. It's value is 12 because it's at the end of all the month.

Reply:
What*